src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke.util;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Represents the parse function of the program. Deals with parsing user input.
+ */
+public class Parser {
+
+    public Parser() {}
```

Comment:
I don't think this constructor is necessary; if I'm not wrong, Java implicitly provides the default constructor for all classes.